src/cpp/security/accesscontrol/Permissions.cpp

```diff
@@ -672,9 +672,9 @@ static bool check_subject_name(const IdentityHandle& ih, AccessPermissionsHandle
                         reader_attributes.plugin_endpoint_attributes = plugin_attributes.mask();
                         writer_attributes.plugin_endpoint_attributes = plugin_attributes.mask();
 
-                        ah->governance_reader_topic_rules_.insert(std::pair<std::string, EndpointSecurityAttributes>(
+                        ah->governance_reader_topic_rules_.push_back(std::pair<std::string, EndpointSecurityAttributes>(
                                     topic_expression, std::move(reader_attributes)));
-                        ah->governance_writer_topic_rules_.insert(std::pair<std::string, EndpointSecurityAttributes>(
+                        ah->governance_writer_topic_rules_.push_back(std::pair<std::string, EndpointSecurityAttributes>(
                                     std::move(topic_expression), std::move(writer_attributes)));
```

Comment:
It seems to me that `reader_attributes` and `writer_attributes` will always have the same values. Perhaps we could have a single collection `governance_topic_rules` ?